### ISSUE TITLE
PSOBB client crashes when killing enemies

### ISSUE THREAD

**Issue by dread-pirate-johnny-spaceboots** [fuzziqersoftware/newserv#141]
I've compiled and got the server up and running and can connect a V2 client and play just fine, but when I connect via the Schthack PSOBB client (dns redirected to the local newserv instance) I can create a character, enter the lobby, create a team, enter the game, but the moment my character kills any enemy I'm immediately disconnected and see the following in the server:

`
I 597 2023-10-14 06:36:57 - [Commands] Received from C-6 (Kahn) (version=BB command=0060 flag=00000000)
0000 | 20 00 60 00 00 00 00 00 3F 06 00 00 00 00 1F 84 |   `     ?
0010 | 01 00 0B 00 2E 19 35 43 20 29 B2 41 08 EE A9 43 |     . 5C ) A   C
I 597 2023-10-14 06:36:57 - [Commands] Received from C-6 (Kahn) (version=BB command=0060 flag=00000000)
0000 | 10 00 60 00 00 00 00 00 43 02 00 00 1F 84 00 00 |   `     C
I 597 2023-10-14 06:36:58 - [Commands] Received from C-6 (Kahn) (version=BB command=0060 flag=00000000)
0000 | 14 00 60 00 00 00 00 00 46 03 00 00 01 00 00 00 |   `     F
0010 | 69 10 00 00                                     | i
I 597 2023-10-14 06:36:58 - [Commands] Received from C-6 (Kahn) (version=BB command=0060 flag=00000000)
0000 | 14 00 60 00 00 00 00 00 0A 03 69 10 69 00 3B 00 |   `       i i ;
0010 | 00 06 B0 20                                     |
I 597 2023-10-14 06:36:58 - [Commands] Received from C-6 (Kahn) (version=BB command=0060 flag=00000000)
0000 | 10 00 60 00 00 00 00 00 44 02 00 00 6D 82 00 00 |   `     D   m
I 597 2023-10-14 06:36:58 - [Commands] Received from C-6 (Kahn) (version=BB command=0060 flag=00000000)
0000 | 14 00 60 00 00 00 00 00 46 03 00 00 01 00 00 00 |   `     F
0010 | 69 10 00 00                                     | i
I 597 2023-10-14 06:36:58 - [Commands] Received from C-6 (Kahn) (version=BB command=0060 flag=00000000)
0000 | 14 00 60 00 00 00 00 00 0A 03 69 10 69 00 3C 00 |   `       i i <
0010 | 00 0A B0 20                                     |
I 597 2023-10-14 06:36:58 - [Commands] Received from C-6 (Kahn) (version=BB command=0060 flag=00000000)
0000 | 14 00 60 00 00 00 00 00 C8 03 69 10 69 00 00 00 |   `       i i
0010 | 01 00 00 00                                     |
W 597 2023-10-14 06:36:58 - [Server] Error processing client command: BB-only command sent in non-BB game
I 597 2023-10-14 06:36:58 - [Server] Client disconnected: C-6 on fd 37
`

Any idea why the server believes this is a "non-bb" game?

**Comment by dread-pirate-johnny-spaceboots** [comment]
Also having issues with the Ephinea PSOBB client. When I override ephinea1.pioneer2.net to point to 127.0.0.1 the client connects to the patch server, i see the newserv patch message, press Enter to continue, I get the "Connection in progress... please wait..." and then the client exits completely. 

All I see in the server log is:
`
I 779 2023-10-15 13:59:36 - [Commands] Sending to C-5 (version=Patch command=12 flag=00)
0000 | 04 00 12 00                                     |
I 779 2023-10-15 13:59:36 - [Server] Client disconnected: C-5 on fd 36
I 779 2023-10-15 13:59:36 - [C-5] Deleted
`

**Comment by dread-pirate-johnny-spaceboots** [comment]
Trying with the Tethealla client available on archive.org lets me connect to the patch server, successfully get into the lobby and create a team, but once I try to enter the game the client crashes.

Messages from the server:
`I 780 2023-10-15 14:29:59 - [Commands] Received from C-6 (Karn) (version=BB command=008A flag=00000000)
0000 | 08 00 8A 00 00 00 00 00                         |
I 780 2023-10-15 14:29:59 - [Commands] Sending to C-6 (Karn) (version=BB command=008A flag=00000000)
0000 | 14 00 8A 00 00 00 00 00 09 00 45 00 6B 00 6A 00 |           E k j
0010 | 68 00 00 00                                     | h
I 780 2023-10-15 14:30:05 - [Commands] Received from C-6 (Karn) (version=BB command=0060 flag=00000000)
0000 | 20 00 60 00 00 00 00 00 3F 06 00 00 00 00 00 C0 |   `     ?
0010 | 0F 00 00 00 CE FE 64 43 00 00 00 00 B8 FF 7D 43 |       dC      }C
I 780 2023-10-15 14:30:05 - [Commands] Received from C-6 (Karn) (version=BB command=0060 flag=00000000)
0000 | 10 00 60 00 00 00 00 00 1F 02 00 00 00 00 00 00 |   `
I 780 2023-10-15 14:30:05 - [Commands] Received from C-6 (Karn) (version=BB command=0060 flag=00000000)
0000 | 0C 00 60 00 00 00 00 00 3B 01 00 00             |   `     ;
I 780 2023-10-15 14:30:05 - [Commands] Received from C-6 (Karn) (version=BB command=006F flag=00000000)
0000 | 08 00 6F 00 00 00 00 00                         |   o
I 780 2023-10-15 14:30:05 - [Commands] Sending to C-6 (Karn) (version=BB command=00B1 flag=00000000)
0000 | 20 00 B1 00 00 00 00 00 32 30 32 33 3A 31 30 3A |         2023:10:
0010 | 31 35 3A 20 31 37 3A 30 30 3A 30 35 2E 30 30 30 | 15: 17:00:05.000
I 780 2023-10-15 14:30:05 - [Commands] Sending to C-6 (Karn) (version=BB command=0095 flag=00000000)
0000 | 08 00 95 00 00 00 00 00                         |
W 780 2023-10-15 14:30:06 - [Server] Client caused error 104 (Connection reset by peer)
I 780 2023-10-15 14:30:06 - [Server] Client disconnected: C-6 on fd 36
I 780 2023-10-15 14:30:06 - [Lobby/15] Deleted lobby
I 780 2023-10-15 14:30:06 - [C-6] Deleted
I 780 2023-10-15 14:30:06 - Saved account data file system/players/account_a.nsa to filesystem
I 780 2023-10-15 14:30:06 - Added 19 seconds to play time
I 780 2023-10-15 14:30:06 - Saved player data file system/players/player_a_1.nsc to filesystem
`

**Comment by fuzziqersoftware** [comment]
There are three separate issues here:

First, BB-only command sent in non-BB game - this was due to a bug I introduced just recently, which I fixed earlier today.

Second, the Ephinea client isn't officially supported on newserv. There is a way to get it to work, but it's very convoluted and it's often better to just use the official Tethealla release.

Third, client crashing - there appears to be an uninitialized memory bug in the client that newserv was triggering when a BB client joins a game, which could cause the client to crash when it tries to use the player's current EXP value. This should be fixed now.
